src/utils/validateLocusCondition.ts

```diff
@@ -0,0 +1,50 @@
+import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
+import dtreeStore from '@store/dtree'
+import filterStore from '@store/filter'
+
+export const validateLocusCondition = (
+  value: string,
+  setIsErrorVisible: (arg: boolean) => void,
+  groupName?: string,
+  currentStepIndex?: number,
+  // eslint-disable-next-line max-params
+): void => {
+  let numberValue = value[3]
+  let lastIndexOfName = 3
+
+  if (+value[4] && value.length > 4) numberValue = value.slice(3, 5)
+
+  if (numberValue) lastIndexOfName = 2 + numberValue.length
+
+  if (
+    value.slice(0, 3) !== 'chr' ||
+    !+numberValue ||
+    +numberValue > 23 ||
+    value[lastIndexOfName + 1] !== ':' ||
+    (!+value.slice(lastIndexOfName + 2) &&
+      value.slice(lastIndexOfName + 2) !== '')
+  ) {
+    setIsErrorVisible(true)
+    filterStore.setError('out of choice')
+  } else {
+    setIsErrorVisible(false)
+    filterStore.setError('')
+
+    if ((currentStepIndex || currentStepIndex === 0) && groupName) {
+      const indexForApi = dtreeStore.getStepIndexForApi(currentStepIndex)
+
+      dtreeStore.setCurrentStepIndexForApi(indexForApi)
+
+      const params = `{"locus":"${value}"}`
+
+      dtreeStore.fetchStatFuncAsync(groupName, params)
+    } else {
+      const params = `{"locus":"${value}"}`
+
+      filterStore.fetchStatFuncAsync(
+        FuncStepTypesEnum.GeneRegion,
+        params || JSON.stringify({ locus: '' }),
```

Comment:
The interpreter never goes to the second part of the expression.
Better to define a single `const params = {locus: value || ''}` outside the if-statements (since you are using `value` from outside the if-scopes) and further pass JSON.stringify(params)